tests/libvmi/vmi.go

```diff
@@ -126,8 +126,10 @@ func WithSEV() Option {
 }
 
 func baseVmi(name string) *kvirtv1.VirtualMachineInstance {
+	defaultTestGracePeriod := int64(0)
 	vmi := kvirtv1.NewVMIReferenceFromNameWithNS("", name)
 	vmi.Spec = kvirtv1.VirtualMachineInstanceSpec{Domain: kvirtv1.DomainSpec{}}
+	vmi.Spec.TerminationGracePeriodSeconds = &defaultTestGracePeriod
```

Comment:
> Users of libvmi do not to know about libvmi.DefaultTestGracePeriod. If
> they care about the grace period of their VMs, they can set it with
> WithTerminationGracePeriod(). Nobody does that, so exporting
> WithTerminationGracePeriod is probably premature.

I do not know what is the meaning of assigning a zero grace period vs keeping it a `nil`. I think the original idea with the default was to copy the existing settings from the code, but now that you want to move it, maybe it is worth asking why we need to set it to zero in the first place.

Also, if you remove all usages of `WithTerminationGracePeriod()`, I see no other option but to remove that builder function as well, otherwise it is left as dead code.
I do find it a bit inaccurate to set this default here so deep. All other settings of the `baseVMI` are a bare minimum... is this the same?

Reply:
> > Users of libvmi do not to know about libvmi.DefaultTestGracePeriod. If
> > they care about the grace period of their VMs, they can set it with
> > WithTerminationGracePeriod(). Nobody does that, so exporting
> > WithTerminationGracePeriod is probably premature.
> 
> I do not know what is the meaning of assigning a zero grace period vs keeping it a `nil`. I think the original idea with the default was to copy the existing settings from the code, but now that you want to move it, maybe it is worth asking why we need to set it to zero in the first place.

The [server-side default is 30s](https://github.com/kubevirt/kubevirt/blob/dd649a1ca34/staging/src/kubevirt.io/api/core/v1/types.go#L39). I assumed that we set it to 0s in the tests because we want the tests to run faster, and we do not care about anything in the test vm to shut down cleanly. Commit 8c72656a52 changed it from 1s to 0s and I think we should keep it.

> 
> Also, if you remove all usages of `WithTerminationGracePeriod()`, I see no other option but to remove that builder function as well, otherwise it is left as dead code.

I agree that it is currently left dead, but we make a strong push to move all code to `libvmi`, and we do have a test for the termination grace period in `tests/vmlifecycle_test.go`. It should not be long until we reintroduce a function like this, hence my reluctance to drop it now. But of course, I'll drop the function if you insist.

> I do find it a bit inaccurate to set this default here so deep. All other settings of the `baseVMI` are a bare minimum... is this the same?

Do you think that it belongs better in `New()`? I don't really mind. I pushed it down because 99% of the tests don't need to know about it.

Reply:
Appending to the `opts` in `New` may make it a bit nicer and cause the function to be actually used.
My main concern here is that we will start seeing other things added this way to that base function, which was not the intention. It will make it messy IMO, as we start adding defaults all over the stack, like the previous version of the VMI building.

Said that, I do not think it is worth overthinking this. It is good enough.

Reply:
> Appending to the `opts` in `New` may make it a bit nicer and cause the function to be actually used. 

Ok, but it needs to be pre-pended, to make future override possible.